php/class-dashboard-widget.php

```diff
@@ -0,0 +1,200 @@
+<?php
+/**
+ * Customize Snapshot Dashboard Event handler.
+ *
+ * @package CustomizeSnapshots
+ */
+
+namespace CustomizeSnapshots;
+
+/**
+ * Class Dashboard_Widget
+ *
+ * @package CustomizeSnapshots
+ */
+class Dashboard_Widget {
+
+	/**
+	 * Snapshot manager.
+	 *
+	 * @var Customize_Snapshot_Manager
+	 */
+	public $manager;
+
+	/**
+	 * Wrong date.
+	 *
+	 * @var int
+	 */
+	public $error_code = 0;
+
+	/**
+	 * Error codes and message mapping.
+	 *
+	 * @var array
+	 */
+	public $error = array();
+
+	/**
+	 * Dashboard_Widget constructor.
+	 *
+	 * @param Customize_Snapshot_Manager $manager manager object.
+	 */
+	public function __construct( $manager ) {
+		$this->manager = $manager;
+		add_action( 'wp_dashboard_setup', array( $this, 'add_widget' ) );
+		add_action( 'load-index.php', array( $this, 'handle_future_snapshot_preview_request' ) );
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_admin_dashboard_scripts' ) );
+
+		$this->error = array(
+			1 => __( 'Please select future date.', 'customize-snapshots' ),
+			2 => __( 'No snapshot found to preview for given date.', 'customize-snapshots' ),
+			3 => __( 'Something went wrong while merging snapshots.', 'customize-snapshots' ),
+		);
+	}
+
+	/**
+	 * Enqueue Dashboard styles.
+	 *
+	 * @param string $hook current page.
+	 */
+	public function enqueue_admin_dashboard_scripts( $hook ) {
+		$handle = 'customize-snapshots-dashboard';
+		if ( 'index.php' === $hook ) {
+			wp_enqueue_style( $handle );
+		}
+	}
+
+	/**
+	 * Add widget.
+	 */
+	public function add_widget() {
+		wp_add_dashboard_widget(
+			'customize_site_state_future_snapshot_preview',
+			__( 'Preview Future Site State', 'customize-snapshots' ),
+			array( $this, 'render_widget' )
+		);
+	}
+
+	/**
+	 * Render widget.
+	 */
+	public function render_widget() {
+		$date_time = current_time( 'mysql' );
+		$date_time = new \DateTime( $date_time );
+		if ( isset( $_POST['year'], $_POST['month'], $_POST['day'], $_POST['hour'], $_POST['minute'], $_REQUEST['_wpnonce'] ) && wp_verify_nonce( $_REQUEST['_wpnonce'], 'customize_site_state_future_snapshot_preview' ) ) {
+			$date_time->setTimestamp( strtotime( "{$_POST['year']}-{$_POST['month']}-{$_POST['day']} {$_POST['hour']}:{$_POST['minute']}" ) );
+		}
+		if ( $this->error_code ) {
+			echo '<p class="error-message">' . esc_html( $this->error[ $this->error_code ] ) . '</p>';
+		} ?>
+		<form method="post">
+				<div class="preview-future-state date-inputs clear">
+					<label>
+						<span class="screen-reader-text"><?php esc_html_e( 'Month', 'customize-snapshots' ); ?></span>
+						<?php $month = Customize_Snapshot_Manager_Compat::get_month_choices(); ?>
+						<select id="snapshot-date-month" class="date-input month" data-date-input="month" name="month">
+							<?php
+							foreach ( $month['month_choices'] as $month_choice ) {
```

Comment:
Since we're embedding PHP in HTML, let's use the alternative control structure: 

```php
<?php foreach ( $month['month_choices'] as $month_choice ) : ?>
```

And then do:

```php
<option value="<?php echo esc_attr( ... ) ?>...
```

Instead of `echo`'ing a string.